### PR TITLE
Support AU v2023.10.24

### PR DESCRIPTION
**Pull request by Tommy-XL**
# Done!

### Progress:
- [x] Update Package Reference #129 
- [x] Fix Build Errors #131
- [x] Change supported version Among Us
- [x] Fix Black Screen in The Fungle
- [x] Add check `SystemTypes.MushroomMixupSabotage` to sabotage activity
- [x] Fix Sabotage Cooldown
- [x] Add setting in Random Maps for The Fungle map
- [x] Fix some bugs during "Mushroom Mixup" sabotage
   - [x] Nickname players (Not need to fix)
   - [x] Fix bug when guardian shield returns all skins
- [x] Fix `ShipStatus.UpdateSystem`
- [x] Fix add-ons in `ShipStatus.UpdateSystem`
   - [x] Fool
   - [x] Unlucky
   - [x] Repairman
- [x] Fix roles in `SwitchSystem` #146 
   - [x] Mechanic
   - [x] Alchemist
   - [x] Fix setting `Block Switches When They Are Up`
- [x] Fix Teleport Players (`RpcTeleport`)#139
- [x] `RpcMurderPlayer` => `RpcMurderPlayerV3` #139 
- [x] Fix kill button for Host
- [x] Fix kill button for Modded Non-Host
- [x] Fix kill button for Vanilla players (Fixed i think)
- [x] Fix Teleport when player uses Zipline
- [x] Fix Kill Flash
- [x] Disable Comms Camouflage on The Fungle map (setting) #145
- [x] Add locations positions for Random Spawn (The Fungle)
- [x] Add The Fungle map in Door Reset
- [x] Fix setting `Fall From Ladders`
- [x] Add setting for "Disable Devices" #138
- [x] Fix devices roles like: #138 
   - [x] Anti Adminer
   - [x] Telecommunication

### New Settings:
- [x] `Disable Spore Trigger (Fungle)`
- [x]  `Disable Zipline (Fungle)`
   - [x] `Disable Zipline From Top`
   - [x] `Disable Zipline From Under`

### CODE REVIEW

**Comment by Ultradragon005** [comment]
ok

**Comment by Tommy-XL** [comment]
Damn, Misklik, lol

**Comment by PHGaming121** [comment]
cultivators level 3 kills (Bomber kills) doesnt work (this is a bug report for git source version)

**Comment by NikoCat233** [comment]
Just bcz rpcmurderplayerv3 doesnt work sometimes....
Send murder calls with decision by host instead

File: main.cs [hunk 1]
```diff
@@ -234,8 +234,8 @@ public class Main : BasePlugin
     public static IEnumerable<PlayerControl> AllPlayerControls => PlayerControl.AllPlayerControls.ToArray().Where(p => p != null);
     public static IEnumerable<PlayerControl> AllAlivePlayerControls => PlayerControl.AllPlayerControls.ToArray().Where(p => p != null && p.IsAlive() && !p.Data.Disconnected && !Pelican.IsEaten(p.PlayerId));
 
-    public static List<PlayerControl> AllPlayerControlsList => (List<PlayerControl>)PlayerControl.AllPlayerControls.ToArray().Where(p => p != null);
-    public static List<PlayerControl> AllAlivePlayerControlsList => (List<PlayerControl>)PlayerControl.AllPlayerControls.ToArray().Where(p => p != null && p.IsAlive() && !p.Data.Disconnected && !Pelican.IsEaten(p.PlayerId));
+    //public static List<PlayerControl> AllPlayerControls => PlayerControl.AllPlayerControls.ToArray().Where(p => p != null).ToList();
+    //public static List<PlayerControl> AllAlivePlayerControls => PlayerControl.AllPlayerControls.ToArray().Where(p => p != null && p.IsAlive() && !p.Data.Disconnected && !Pelican.IsEaten(p.PlayerId)).ToList();
```
Comment (Gurge44) [inline main.cs:239]: Yes, this works

Comment (Tommy-XL) [reply main.cs:239]: I've temporarily commented this, out as I'll be working on it a little later

**Comment by Hyz-sui** [comment]
Hey, I'm too busy fixing TOH to send a pull request, so I'll leave a hint to fix `ISystemType` like SwitchSystem, you should use `MessageReader.Get`
https://github.com/Hyz-sui/TownOfHost-H/commit/c3ee76774ef4fbc4e9a7eeb06815d3855ad71624#diff-9f31ec8bb8519215d7c784fdb78965993b82c23f7340a6738d408aff52c78064R60-R71
hope this helps

**Comment by Tommy-XL** [comment]
> Hey, I'm too busy fixing TOH to send a pull request, so I'll leave a hint to fix `ISystemType` like SwitchSystem, you should use `MessageReader.Get` [Hyz-sui/TownOfHost-H@c3ee767#diff-9f31ec8bb8519215d7c784fdb78965993b82c23f7340a6738d408aff52c78064R60-R71](https://github.com/Hyz-sui/TownOfHost-H/commit/c3ee76774ef4fbc4e9a7eeb06815d3855ad71624#diff-9f31ec8bb8519215d7c784fdb78965993b82c23f7340a6738d408aff52c78064R60-R71) hope this helps


Thanks! We're currently working on a fix for the `SwitchSystem`

File: Patches/SabotageSystemPatch.cs [hunk 2]
```diff
@@ -97,6 +138,64 @@ public static void Initialize()
...
+
+            case CustomRoles.Sidekick when Jackal.CanUseSabotageSK.GetBool():
+                return true;
+
+            case CustomRoles.Bandit when Bandit.CanUseSabotage.GetBool():
+                return true;
+
+            case CustomRoles.Glitch:
+                Glitch.Mimic(player);
+                return true;
```
Comment (Gurge44) [inline Patches/SabotageSystemPatch.cs:181]: return false here cause the sabotage is just a trigger for mimic, Glitch cannot actually sabotage

Comment (Tommy-XL) [reply Patches/SabotageSystemPatch.cs]: Fixed

**Comment by Tommy-XL** [comment]
@Hyz-sui Have you found a way to somehow track the use of Zipline on the player so that it eventually returns false or true, for example like here:

image

**Comment by Hyz-sui** [comment]
It seems that `inMovingPlat` is being diverted(`ZiplineBehaviour.PreparePlayerForZipline`)

**Comment by Tommy-XL** [comment]
> is being diverted

Oh, thx, I thought that `inMovingPlat` only tracks the platform on Airship, lol

**Review by Tommy-XL — COMMENTED** [review]
All progress of support for AU v2023.10.24 seems to be completed (there may be small bugs that we did not notice)

But, I would like to thank people for their help with the code and for their assistance in development in particular:
@NikoCat233 @Gurge44 @Hyz-sui @Huier-Huang @FuroYT
You are amazing!

**Review by 0xDrMoe — APPROVED** [review]

**Pull request merged**